src/App.tsx

```diff
@@ -1,61 +1,104 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/TodoInfo';
+
+function findUser(id: number) {
+  return usersFromServer.find(user => user.id === id) || null;
+}
+
+const preparedUsers: Todo[] = todosFromServer.map(todo => {
+  const userFound = usersFromServer.find(userFromServer => (
+    todo.userId === userFromServer.id));
+
+  return {
+    ...todo,
+    user: userFound || null,
+  };
+});
 
 export const App = () => {
+  const [todos, setTODO] = useState<Todo[]>(preparedUsers);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const addNewTODO = (todoTitle: string, userId: number) => {
+    const newTODO: Todo = {
+      id: Date.now(),
+      title: todoTitle,
+      userId,
+      completed: false,
+      user: findUser(selectedUserId),
+    };
+
+    setTODO((previousTODO) => [...previousTODO, newTODO]);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title && selectedUserId) {
+      addNewTODO(title, selectedUserId);
+      setTitle('');
+      setSelectedUserId(0);
+    }
+
+    if (!title) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrorUser(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleFormSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={title}
+            onChange={event => setTitle(event.target.value)}
+          />
+          {errorTitle
+          && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <select
+            data-cy="userSelect"
+            value={selectedUserId}
+            onChange={event => setSelectedUserId(+event.target.value)}
+          >
             <option value="0" disabled>Choose a user</option>
+            {usersFromServer.map(user => (
+              <option value={user.id}>{user.name}</option>
```

Comment:
Include 'key' prop when rendering list items. Use the 'user.id' as the key when iterating through usersFromServer 🗝️